clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -613,7 +602,24 @@ private KafkaConsumer(ConsumerConfig config,
     }
 
     /**
-     * Incrementally subscribes to the given list of topics and uses the consumer's group management functionality
+     * Get the current subscription. Will return the same topics used in the most recent call to
+     * {@link #subscribe(List, ConsumerRebalanceListener)}, or an empty set if no such call has been made.
+     * @return The set of topics currently subscribed to
+     */
+    public Set<String> subscription() {
+        acquire();
+        try {
+            return Collections.unmodifiableSet(this.subscriptions.subscription());
```

Comment:
Yep, makes perfect sense. I think we should just return a copy instead of a view into the underlying collection. I don't know if users get any real value from the latter. Seems more likely to cause confusion.